python/hsml/core/model_api.py

```diff
@@ -143,3 +144,87 @@ def delete(self, model_instance):
             model_instance.id,
         ]
         _client._send_request("DELETE", path_params)
+
+    def set_tag(self, model_instance, name, value):
+        """Attach a name/value tag to a model.
+
+        A tag consists of a name/value pair. Tag names are unique identifiers.
+        The value of a tag can be any valid json - primitives, arrays or json objects.
+
+        :param path: path to add the tag
+        :type path: str
+        :param name: name of the tag to be added
+        :type name: str
+        :param value: value of the tag to be added
+        :type value: str
```

Comment:
Is the value expected to be the JSON string? We may want to check the type `value` before calling `json.dumps()` to avoid serializing twice.

Reply:
Yeah it's expected to be a JSON string, I will add a check.